### ISSUE TITLE
Editor hangs when inserting long lines of text

### ISSUE THREAD

**Issue by Hagtronics** [spyder-ide/spyder#19242]
### What steps will reproduce the problem?

1 - Open Spyder.
2 - File -> New File.
3 - Delete what is in the new file.
4 - Open a text file in an another editor that has a very long line, like a csv file with no line breaks. 
5 - Copy the long line from the other editor.
6 - Paste this line into the Spyder Editor.
7 - Spyder will hang.

I noticed that 1000 characters long line is OK for pasting. 10,000 Characters or more hangs Spyder Editor. I don't know where the exact line length between working and not working is however.

I agree that this is uncommon usage - but I was using the Spyder editor to make a data file as it would have been useful for me. 

A better behavior IMHO would be to throw a message that says: "Line too long" and truncate the line instead of hanging. 

Feel free to say: "You're an idiot" and close the ticket. This is an oddball situation. I just thought I would mention it.

## Versions

* Spyder version: 5.3.2  (standalone)
* Python version: 3.8.10 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                 :  1.4.1 (OK)
chardet >=2.0.0                      :  5.0.0 (OK)
cloudpickle >=0.5.0                  :  2.1.0 (OK)
cookiecutter >=1.6.0                 :  2.1.1 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree                         :  None (OK)
IPython >=7.31.1;<8.0.0              :  7.34.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.1 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.5.1 (OK)
keyring >=17.0.0                     :  23.6.0 (OK)
nbconvert >=4.0                      :  6.5.0 (OK)
numpydoc >=0.6.0                     :  1.4.0 (OK)
paramiko >=2.4.0                     :  2.11.0 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.1 (OK)
pygments >=2.0                       :  2.12.0 (OK)
pylint >=2.5.0;<3.0                  :  2.14.4 (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0                 :  1.5.0 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.1.10                   :  0.2.1 (OK)
qtawesome >=1.0.2                    :  1.1.1 (OK)
qtconsole >=5.3.0;<5.4.0             :  5.3.1 (OK)
qtpy >=2.1.0                         :  2.1.0 (OK)
rtree >=0.9.7                        :  1.0.0 (OK)
setuptools >=49.6.0                  :  63.1.0 (OK)
sphinx >=0.6.6                       :  5.0.2 (OK)
spyder_kernels >=2.3.2;<2.4.0        :  2.3.2 (OK)
textdistance >=4.2.0                 :  4.3.0 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog                             :  2.1.9 (OK)
zmq >=22.1.0                         :  23.2.0 (OK)

# Optional:
cython >=0.21                        :  0.29.30 (OK)
matplotlib >=3.0.0                   :  3.5.2 (OK)
numpy >=1.7                          :  1.22.4 (OK)
pandas >=1.1.1                       :  1.4.3 (OK)
scipy >=0.17.0                       :  1.8.1 (OK)
sympy >=0.7.3                        :  1.10.1 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)
```

**Comment by dalthviz** [comment]
Hi @Hagtronics thank you for the feedback! Could you share with us an example file from where you were copying the long line, please? Is this hanging always happening when copying the long line?

Also, Spyder 5.3.3 was release recently, could you update and run Spyder on debug mode? For that you can run from a cmd something like (replacing the path used to install Spyder or your user directory in case the default path was used):

```cmd
C:\Users\<your user directory>\AppData\Local\Programs\Spyder\Python\python.exe "C:\Users\<your user directory>\AppData\Local\Programs\Spyder\Spyder.launch.pyw"  --debug-info verbose
```

Maybe we can get, from the debug log, some info about what is causing the freezing issue.

Any new info in order to replicate/understand better what could be happening is greatly appreciated!

**Comment by Hagtronics** [comment]
Hello - OK, I upgraded to 5.3.3 and the problem is gone. I could paste a line of 100,000 characters and it took a few seconds, but the program never went 'White Screen' and finally responded very well behaved.

Case closed, sorry to waste your time.

**Comment by ccordoba12** [comment]
That's great news! Thanks @Hagtronics for letting us know about it because we were not aware that the fixes we did in 5.3.3 to solve some performance issues in the editor would also solve this problem.
